tools/perf/ucc_pt_coll_alltoall.cc

```diff
@@ -87,17 +89,22 @@ ucc_pt_coll_alltoall::ucc_pt_coll_alltoall(ucc_datatype_t dt,
                                   &mem_map_params, &dst_memh_size, &dst_memh),
                       exit, st);
 
+        // Synchronize memh size across all ranks to ensure consistent buffer allocation
+        dst_memh_size_dbl = (double)dst_memh_size;
+        comm->allreduce(&dst_memh_size_dbl, &dst_memh_size_max_dbl, 1, UCC_OP_MAX);
```

Comment:
pls update comm->allreduce to work with different datatypes

Reply:
updated